docker-compose.yml

```diff
@@ -1,10 +1,32 @@
 version: '3.8'
 
-
 services:
-  gatsby:
-    build: ./
+
+  # Run the gatsby server
+  # Usage: docker-compose up develop
+  develop:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    command: gatsby develop -H 0.0.0.0
+    volumes:
+      - .:/app # allows sharing of local source files for hot-reloading and gatsby notifs
+      - node_modules:/app/node_modules # isolate and preserve node modules in container between builds
+      - cache:/app/.cache # isolate and preserve container gatsby cache
+      - public:/app/public # isloate and preserve container public artifacts
```

Comment:
@lelia I noticed you removed `volumes` -- the way it was written only worked from a fresh / clean install.

While that might be a reasonable assumption in most cases (as you would either run yarn locally, or run it using docker) it clearly caused enough of an issue in this case that it needed fixing. I've added these lines and want to make sure we both agree it's important. Without volume mounting we _cannot_ do the live reloading, live editing expected behavior folks would expect from a modern web server. One would have to build a new container with every edit, if we don't mount these volumes.

Let me know what you think!

Reply:
Yup, not sure if u saw my note at the bottom of the PR about it - 

> **NOTE:** I was unable to find a way to keep the volumes section in docker-compose.yml without running into subsequent yarn build errors, even when including COPY instructions within the Dockerfile and accounting for relative/absolute pathing, etc. So, the configuration as-is will likely not support hot-reloads, but we could always file this as a future enhancement and still get some degree of Docker functionality in the meantime.

I sunk several hours into trying all sorts of ways to make it work before time-boxing it and figuring it would be better to discuss once you were back to see if you had any ideas / obvious things I was missing. My preference is always to use `volumes`, so if the changes you just added allow it to work without hitting the installation errors I kept getting I'm all for it! 🚀

Reply:
I think it does fix it -- It worked locally as I tried it. Feel free to give it a whirl and see if it isn't just my machine.

Reply:
Confirmed hot-reloading works with new `volumes` config -- thanks!

Also verified that `docker-compose up` still works on a completely clean install (eg. with `node_modules`, `.cache` and `public` rm -rf'd from the local repo context). 

:shipit: 🎉